examples/rekey-example/README.md

```diff
@@ -0,0 +1,56 @@
+# Simple Round-Trip Example
+
```

Comment:
I think it would be good to concisely define what rekeying is here.

Reply:
I gave it a shot. Let me know if you have suggestions